[FRONT-4] Refactor frontend

cnk1dev opened the pull request:
### Description of issue

- [x] Replace template string classNames with `clsx` + `twMerge`
- [x] Improve Tailwind config (colors, font sizes, border radius)
- [x] Add prettier config for ui package with auto import sorting + tailwind className sorting
- [x] Run `prettier:fix` on all files

### How I implemented

Did all of the steps above, maybe we need to update the PR verify rules because sometimes it doesn't make sense to write a long description.

cnk1dev committed:
Update favicon

cnk1dev committed:
Remove unnecessary custom Tailwind font sizes

cnk1dev committed:
Add custom border radius to Tailwind config

cnk1dev committed:
Add prettier config for ui package, install prettier + tailwind helper plugins, run prettier:fix on all files

pczern commented:
*Important*
Why do you need a different prettier config
Please explain Design choices of removing `defaultProps` and how and why tailwind config is updated

cnk1dev commented:
> _Important_ Why do you need a different prettier config Please explain Design choices of removing `defaultProps` and how and why tailwind config is updated

- As stated in the PR description, the updates to the prettier config add auto import sorting and tailwind className sorting in the ui package
- I removed `defaultProps` because it's also possible and much cleaner to set default values for parameters using TypeScript. Also, almost all of the default values we had were either `undefined` or `null`, which doesn't add any value if all component props are already properly typed
- I removed the custom font sizes from the Tailwind config which were just duplicates of the existing ones and added rounded-[5px] which was used everywhere to the config

cnk1dev commented:
Moved a few of the commits (including the `defaultProps` topic) to another PR to make reviewing easier because this PR has 90+ changed files due to autoformatting

pczern committed:
Merge pull request #35 from viets-software-club/staging

First Production Environment with Helm

cnk1dev committed:
Merge branch 'staging' into chore/front-4-refactor-frontend

cnk1dev committed:
Merge branch 'staging' into chore/front-4-refactor-frontend

cnk1dev committed:
Fix package-lock.json + prettier issues
